### PR TITLE
Update README and Developer Guide

### PR DESCRIPTION
**Pull request by IanHoang**
### Description
Shorten readme and improve developer guide

### Issues Resolved
N/A

### Testing
N/A

[Describe how this change was tested]

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -2,30 +2,46 @@
...
 
 ### Setup
 
-Use the following command-line instructions to set up OpenSearch Benchmark for development:
+To develop OSB properly, it is recommended that you fork the official OpenSearch Benchmark repository.
+
+After you git cloned the forked copy of OpenSearch Benchmark, use the following command-line instructions to set up OpenSearch Benchmark for development:
 ```
-git clone https://github.com/opensearch-project/OpenSearch-Benchmark
 cd OpenSearch-Benchmark
```
Comment (gkamat) [inline DEVELOPER_GUIDE.md:45]: It may be better to use lowercase for consistency, since the GitHub project name is in lowercase.  Else, there can be issues on case-sensitive systems if there is already a directory in the wrong case.

File: DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -80,6 +94,7 @@ Integration tests can be run on the following operating systems:
   * Amazon Linux 2
   * MacOS
 
+Invoke integration tests by running the following command within the root directory of the repository:
```
Comment (gkamat) [inline DEVELOPER_GUIDE.md:97]: Would be good to note the amount of time the user should expect to run the integ tests.

File: osbenchmark/results_publisher.py [hunk 3]
```diff
@@ -130,6 +130,8 @@ def publish(self):
 
         for record in stats.op_metrics:
             task = record["task"]
+            self.logger = logging.getLogger(__name__)
+            self.logger.info("Record: [%s], Task: [%s]", record, task)
```
Comment (gkamat) [inline osbenchmark/results_publisher.py:134]: Might be better to have this in a separate commit.

**Review by gkamat — APPROVED** [review]
Minor changes suggested -- but they can be addressed in a future change.

File: osbenchmark/results_publisher.py (hunk 3 above)
Comment (IanHoang) [reply osbenchmark/results_publisher.py:134]: Good catch, let me remove this!

File: DEVELOPER_GUIDE.md (hunk 1 above)
Comment (IanHoang) [reply DEVELOPER_GUIDE.md]: Good catch!
